### PR TITLE
refactor(client): ts-migration of i18n/validate-keys.js

### PR DESCRIPTION
**Pull request by foolhardy21**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes part of #43778

### CODE REVIEW

**Comment by foolhardy21** [comment]
Hi, please review the code and let me know if this pr has any issues. I would love to contribute better.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Hi @foolhardy21 thanks for putting this together.  Unfortunately, none of `client/i18n` is getting compiled by TypeScript, yet, so the script that calls this `npm run validate-kays` (in the client package.json) will fail.

I think this could be solved by moving validate-keys.ts into `tools/scripts/lint`

@ShaunSHamilton @nhcarrigan any thoughts on this?  The main thing is we don't want another tsconfig.

**Comment by foolhardy21** [comment]
So, should I move the file to `tools/scripts/lint`? Also, do I need to update the validate-keys script in `package.json'?

**Comment by foolhardy21** [comment]
Hi @ojeytonwilliams , I have moved the file, although I have left `npm run validate-keys` in `client/package.json` untouched. please let me know.

File: tsconfig.json
Comment (ojeytonwilliams) [inline tsconfig.json:8]:
```suggestion
```
Sorry, @foolhardy21 I forgot that the relevant `tsconfig.json` is inside `tools/scripts/build`.  The thing to do is *probably*, move the tsconfig to `tools/` and update the scripts accordingly.

**Review by ojeytonwilliams — COMMENTED** [review]
I'd wait until Nick and/or Shaun have time to take a look, but if we move `validate-keys` the package.json that uses it will need updating.

File: tsconfig.json
Comment (foolhardy21) [reply tsconfig.json]: hey @ojeytonwilliams, I have moved the `tsconfig.json` from `tools/scripts/build` to `tools` and also updated the `npm run validate-keys` in `client/package.json`. Due to this, I had to update the project in `.eslintrc.json` and the `lint:ts` testing script in the root 'package.json`. I have added `validate-keys.js` to `.gitignore` and `.eslintignore`. Let me know about further updates. thanks.

**Comment by foolhardy21** [comment]
hey @ojeytonwilliams, can you please review the changes?

File: .gitignore
Comment (ojeytonwilliams) [inline .gitignore:171]:
```suggestion
tools/tsconfig.tsbuildinfo
```

File: client/package.json
Comment (ojeytonwilliams) [inline client/package.json:33]:
```suggestion
    "validate-keys": "tsc -p ../tools/ && node ../tools/scripts/lint/validate-keys.js"
```

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Looking good @foolhardy21, there are just a couple of issues outstanding:

**Comment by foolhardy21** [comment]
I've made the changes @ojeytonwilliams.

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

Nice work @foolhardy21 :100:

**Comment by foolhardy21** [comment]
thanks for helping. shall I resolve the conversations or wait for further approval?

**Comment by ojeytonwilliams** [comment]
@foolhardy21 generally speaking, we like maintainers to resolve conversations, so we can be sure that the issue has been dealt with.  It all looks good to me, so I went ahead and resolved them.

Now it's just a matter of waiting for a second review, in case I missed something.

**Comment by foolhardy21** [comment]
got it. cool.

**Review by naomi-lgbt — APPROVED** [review]
This looks good to me.

**Comment by nhcarrigan** [comment]
Congrats on your first pull request (PR)! 🎉

Thank you for your contribution to the page! 👍
We are happy to accept these changes and look forward to future contributions. 📝

**Pull request merged**
